UKHomeOffice/cerberus-service
# COP-11580 - Task versions accordion Bug

## Description
This PR address the following:
1. Task details being re-rendered in excess due to state changes in context which then causes all child components to rerender.
2. Task details accordion controls being duplicated multiple times due to the re-rendering issue above.
3. Duplicate network call which is triggered by a rogue `useEffect` hook.

## Note
Opted for `useRef` instead of `useEffect` in context as after some investigations, setting state in context also triggers a rerender.
 

## To Test
- Pull & run against dev
- View a task in task details
- Manually refresh the page by method of choice, the accordion controls should not be duplicated.
- On each page reload, clicking on the task versions header (also for multiple versions) should open and close the accordion.
- Can also reload the page multiple times and check accordion.
